README.md

```diff
@@ -93,72 +73,157 @@ npm install
 
 ## Usage
 
-### Generating a new key pair
+### Installing Support for Key Types
 
-Ed25519:
+In order to use this library, you will need to import and install driver
+libraries for key types you'll be working with via the `use()` method.
+
+For example, to use this library with only the `ed25519` key type:
 
 ```js
-const {Ed25519KeyPair} = require('crypto-ld');
+import {CryptoLD} from 'crypto-ld';
+const cryptoLd = new CryptoLD();
+import Ed25519 from 'ed25519-key-pair';
+
+cryptoLd.use(Ed25519);
 
-const keyPair = await Ed25519KeyPair.generate();
+// With only one key type installed, you do not need to specify key type for
+// most operations
+const keyPair = await cryptoLd.generate(); // generates an ed25519 key pair
```

Comment:
This text and example needs to be adjusted to match the code and tests that don't have this default behavior.

Reply:
@davidlehn - I'm not sure I understand.. say more?

Reply:
Calling `generate()` without any options will throw an error.  You have to pass in `type` or `suite`.  This example should be updated to reflect that.

(And if the API really did match this example, I would argue that is a bad idea.  Better to be explicit for things like this to avoid potential bugs.)

Reply:
@davidlehn ah right, thank you, good catch. I'll update the API to reflect that. (I don't think this is a bad idea, I think it's important to streamline the most common use case (which is, using it with just a single suite)).

Reply:
Wait... I'm saying that this call should be explicit.  That default idea in the comment would lead to bugs and failures.  For instance, if you need to support another suite and add a second `use(...)` call, every call that was expecting `generate()` to work would start failing.  That could happen indirectly in some other library you pull in which would be confusing.  Similar issues if you switch the `use(...)` call to another suite, you might have code unexpectedly using wrong algorithms.

Reply:
@davidlehn no prob. I'll change it so that `generate()` always requires an explicit param.